Incompatible with Kotlin 1.7.0-RC

`ErrorType` has moved to a new package - `org.jetbrains.kotlin.types.error.ErrorType`. I can send a PR if you want

```
e: java.lang.NoClassDefFoundError: org/jetbrains/kotlin/types/ErrorType
	at com.squareup.anvil.compiler.internal.DescriptorUtilsKt.argumentType(DescriptorUtils.kt:40)
	at com.squareup.anvil.compiler.internal.reference.AnnotationArgumentReference$Descriptor.toClassReference(AnnotationArgumentReference.kt:331)
	at com.squareup.anvil.compiler.internal.reference.AnnotationArgumentReference$Descriptor.access$toClassReference(AnnotationArgumentReference.kt:296)
	at com.squareup.anvil.compiler.internal.reference.AnnotationArgumentReference$Descriptor$value$2.invoke$parseConstantValue(AnnotationArgumentReference.kt:310)
	at com.squareup.anvil.compiler.internal.reference.AnnotationArgumentReference$Descriptor$value$2.invoke(AnnotationArgumentReference.kt:327)
	at kotlin.UnsafeLazyImpl.getValue(Lazy.kt:81)
	at com.squareup.anvil.compiler.internal.reference.AnnotationArgumentReference$Descriptor.getValue(AnnotationArgumentReference.kt:303)
	at com.squareup.anvil.compiler.internal.reference.AnnotationArgumentReference.value(AnnotationArgumentReference.kt:61)
	at slack.anvil.injection.compiler.AnvilInjectorGenerator.mapKeyType(AnvilInjectorGenerator.kt:212)
	at slack.anvil.injection.compiler.AnvilInjectorGenerator.mapAnnotation(AnvilInjectorGenerator.kt:199)
	at slack.anvil.injection.compiler.AnvilInjectorGenerator.access$mapAnnotation(AnvilInjectorGenerator.kt:52)
	at slack.anvil.injection.compiler.AnvilInjectorGenerator$generateCode$1.invoke(AnvilInjectorGenerator.kt:69)
	at slack.anvil.injection.compiler.AnvilInjectorGenerator$generateCode$1.invoke(AnvilInjectorGenerator.kt:69)
	at kotlin.sequences.TransformingSequence$iterator$1.next(Sequences.kt:210)
	at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:170)
	at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:194)
	at kotlin.sequences.TransformingSequence$iterator$1.hasNext(Sequences.kt:214)
	at kotlin.sequences.SequencesKt___SequencesKt.toCollection(_Sequences.kt:7[87](https://github.com/tinyspeck/slack-android-ng/runs/6697758687?check_suite_focus=true#step:7:89))
	at kotlin.sequences.SequencesKt___SequencesKt.toMutableList(_Sequences.kt:817)
	at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:808)
	at slack.anvil.injection.compiler.AnvilInjectorGenerator.generateCode(AnvilInjectorGenerator.kt:148)
	at com.squareup.anvil.compiler.codegen.CodeGenerationExtension.analysisCompleted$generateCode(CodeGenerationExtension.kt:104)
	at com.squareup.anvil.compiler.codegen.CodeGenerationExtension.analysisCompleted(CodeGenerationExtension.kt:139)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM$analyzeFilesWithJavaIntegration$2.invoke(TopDownAnalyzerFacadeForJVM.kt:115)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:125)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:99)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:264)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:55)
	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:115)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:255)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:101)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:60)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:157)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:52)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:[94](https://github.com/tinyspeck/slack-android-ng/runs/6697758687?check_suite_focus=true#step:7:96))
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:43)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1642)
	at jdk.internal.reflect.GeneratedMethodAccessor120.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:360)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:1[96](https://github.com/tinyspeck/slack-android-ng/runs/6697758687?check_suite_focus=true#step:7:98))
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:587)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:705)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:3[99](https://github.com/tinyspeck/slack-android-ng/runs/6697758687?check_suite_focus=true#step:7:101))
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:704)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:[113](https://github.com/tinyspeck/slack-android-ng/runs/6697758687?check_suite_focus=true#step:7:115)6)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ClassNotFoundException: org.jetbrains.kotlin.types.ErrorType
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:587)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 54 more
```

--- a/compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/DescriptorUtils.kt
+++ b/compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/DescriptorUtils.kt
@@ -13,9 +13,9 @@ import org.jetbrains.kotlin.name.ClassId
 import org.jetbrains.kotlin.name.FqName
 import org.jetbrains.kotlin.resolve.constants.ConstantValue
 import org.jetbrains.kotlin.resolve.constants.KClassValue.Value.NormalClass
-import org.jetbrains.kotlin.types.ErrorType
 import org.jetbrains.kotlin.types.KotlinType
 import org.jetbrains.kotlin.types.TypeUtils
+import org.jetbrains.kotlin.types.isError
 
 // When the Kotlin type is of the form: KClass<OurType>.
 @ExperimentalAnvilApi
@@ -37,7 +37,7 @@ public fun KotlinType.classDescriptor(): ClassDescriptor {
 @ExperimentalAnvilApi
 public fun ConstantValue<*>.argumentType(module: ModuleDescriptor): KotlinType {
   val argumentType = getType(module).argumentType()
-  if (argumentType !is ErrorType) return argumentType
+  if (!argumentType.isError) return argumentType
 
   // Handle inner classes explicitly. When resolving the Kotlin type of inner class from
   // dependencies the compiler might fail. It tries to load my.package.Class$Inner and fails

